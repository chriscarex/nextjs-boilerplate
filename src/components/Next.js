import Link from 'next/link';
import { ROUTE_HOME } from '../constants/routes';

const Next = () => (
  <div>
    <h1>
      Welcome to the next page!
    </h1>

    <div>
      <Link href={ROUTE_HOME}>
        <a>Click here to go to the previous page</a>
      </Link>
    </div>

  </div>
);

export default Next;
