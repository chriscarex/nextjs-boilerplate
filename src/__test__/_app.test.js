import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MyApp from '../pages/_app';
import Next from '../pages/next';

const setup = (pageProps) => {
  const props = {
    Component: Next,
    pageProps
  };

  const component = shallow(<MyApp {...props} />);

  return {
    component,
    props
  };
};

describe('_app', () => {
  it('should match snapshot ', () => {
    const {
      component
    } = setup({
      success: false
    });

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match snapshot if error occurs', () => {
    const {
      component
    } = setup({
      error: { statusCode: 400 }
    });

    expect(toJson(component)).toMatchSnapshot();
  });
});
