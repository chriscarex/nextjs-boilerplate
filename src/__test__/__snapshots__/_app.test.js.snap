// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_app should match snapshot  1`] = `
<Fragment>
  <HeaderComponent />
  <NextPage
    success={false}
  />
  <FooterComponent />
</Fragment>
`;

exports[`_app should match snapshot if error occurs 1`] = `
<Fragment>
  <HeaderComponent />
  <ErrorPage
    statusCode={400}
  />
  <FooterComponent />
</Fragment>
`;
